implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -265,6 +267,21 @@ private void verifyTimeToLive(JWTAuthContextInfo authContextInfo, JwtClaims clai
         }
     }
 
+    private void verifyRequiredClaims(JWTAuthContextInfo authContextInfo, JwtClaims claimsSet) throws ParseException {
+        final Set<String> requiredClaims = authContextInfo.getRequiredClaims();
+
+        if (requiredClaims != null) {
+            final String missingClaims = requiredClaims.stream()
```

Comment:
H @radcortez Nearly there :-), as I implied, lets avoid the stream API. I still think the simple keyset check would suffice.
If the check fails then there must be a `Collection` function which would return the keys which are not intersecting - though  we don't really need to report back what exactly is required - this is usually avoided in the security flows to keep the attackers guessing.

 
It should be `InvalidJwtException` internally - because it is invalid in this case, and then it will be wrapped into a `ParseException` and propagated.

Reply:
Hum, but doing a `keySet().containsAll` and then another `Collection` function to retrieve the keys that not intersect, isn't more inefficient than using the `Stream` stream does everything in a single pass of the date, while in here we need to perform at least 2 iterations.

Plus, I've seen the use of the `Stream` API in Quarkus before :)

Reply:
@radcortez That another `Collection` function can be dropped, as I said reporting what is required back is not needed. Or if you'd really like, if the key set checks fails and debug mode is enabled, run the extra collection function there and log what is missing but still report the opaque enough message back :-)
I'm sure Quarkus has plenty of this code but in this case it is a runtime time, smallrye-jwt is also used in WildFly among other containers, and Darran has already provided one PR to get rid of some stream api so lets continue in this spirit
thanks

Reply:
Ok. Thanks. Just wanted to clarify the reason.

I think we should always have this information for debug purposes, or it is usually a pain to figure out why a token is valid / invalid. (Ops someone changed the configurations and we are now requiring claim X :) )